### PR TITLE
feat: 1368 authorization for curators

### PR DESCRIPTION
**Pull request by ebezzi**
### Reviewers
**Functional:** 
@MDunitz 

**Readability:** 
@seve 

---

Ticket [#1368 ](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1368)

## Changes
- Changes all the collections endpoints to behave differently depending on the user's permissions

## Definition of Done (from ticket)
- Curators can view all the private collections, even if they belong to other users
- Curators can delete any private collection
- Curators can update any private collection
- Curators can start revisions on every private collection.

## Known issues
Verify if this doesn't introduce new authentication issues that might impact non logged in users.

### CODE REVIEW

File: backend/corpora/lambdas/api/v1/collection.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from backend.corpora.lambdas.api.v1.authorization import has_scope
```
Comment (leslieypark) [inline backend/corpora/lambdas/api/v1/collection.py:1]: Super nit: Can we move this import down, after the 3rd party imports?

File: tests/unit/backend/corpora/api_server/test_v1_collection.py [hunk 2]
```diff
@@ -863,3 +863,54 @@ def test__update_collection_links__OK(self):
...
+        collection = self.generate_collection(
+            self.session, visibility=CollectionVisibility.PRIVATE.name, owner="another_test_user_id"
+        )
+
+        headers = {"host": "localhost", "Content-Type": "application/json", "Cookie": get_auth_token(self.app)}
+        test_url = furl(path=f"/dp/v1/collections/{collection.id}", query_params=dict(visibility="PRIVATE"))
+        response = self.app.get(test_url.url, headers=headers)
+
+        # This will pass even for non curators.
+        # Why are users allowed to view private collections that they don't own?
```
Comment (leslieypark) [inline tests/unit/backend/corpora/api_server/test_v1_collection.py:880]: Is this a todo question?

**Review by leslieypark — APPROVED** [review]
🚀  Super nitpicks, but otherwise LGTM

File: tests/unit/backend/corpora/api_server/test_v1_collection.py (hunk 2 above)
Comment (ebezzi) [reply tests/unit/backend/corpora/api_server/test_v1_collection.py:880]: Yes, if you look at the code [here](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/corpora/lambdas/api/v1/collection.py#L39), there is no explicit filter for this endpoint that requires the user to be an owner. Only tombstone datasets and `access_type` are gated. @MDunitz do you know if this is as expected?

Comment (MDunitz) [reply tests/unit/backend/corpora/api_server/test_v1_collection.py:880]: At the time of implementation we decided that private datasets shouldnt be displayed on the portal page but they should be available to anyone with the link to handle cases where the owner of the dataset wanted to share it with other collaborators

**Pull request merged**